cmd/server/pactasrv/initiative_invitation.go

```diff
@@ -0,0 +1,118 @@
+package pactasrv
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/RMI/pacta/cmd/server/pactasrv/conv"
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/oapierr"
+	api "github.com/RMI/pacta/openapi/pacta"
+	"github.com/RMI/pacta/pacta"
+	"go.uber.org/zap"
+)
+
+// Creates an initiative invitation
+// (POST /initiative-invitation)
+func (s *Server) CreateInitiativeInvitation(ctx context.Context, request api.CreateInitiativeInvitationRequestObject) (api.CreateInitiativeInvitationResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	ii, err := conv.InitiativeInvitationFromOAPI(request.Body)
+	if err != nil {
+		return nil, err
+	}
+	id, err := s.DB.CreateInitiativeInvitation(s.DB.NoTxn(ctx), ii)
+	if err != nil {
+		return nil, oapierr.Internal("failed to create initiative invitation", zap.Error(err))
+	}
+	if id != ii.ID {
+		return nil, oapierr.Internal(
+			"failed to create initiative invitation: ID mismatch",
+			zap.String("requested_id", string(ii.ID)),
+			zap.String("actual_id", string(id)),
+		)
+	}
+	return api.CreateInitiativeInvitation204Response{}, nil
+}
+
+// Deletes an initiative invitation by id
+// (DELETE /initiative-invitation/{id})
+func (s *Server) DeleteInitiativeInvitation(ctx context.Context, request api.DeleteInitiativeInvitationRequestObject) (api.DeleteInitiativeInvitationResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	err := s.DB.DeleteInitiativeInvitation(s.DB.NoTxn(ctx), pacta.InitiativeInvitationID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to delete initiative invitation", zap.Error(err))
+	}
+	return api.DeleteInitiativeInvitation204Response{}, nil
+}
+
+// Returns the initiative invitation from this id, if it exists
+// (GET /initiative-invitation/{id})
+func (s *Server) GetInitiativeInvitation(ctx context.Context, request api.GetInitiativeInvitationRequestObject) (api.GetInitiativeInvitationResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	ii, err := s.DB.InitiativeInvitation(s.DB.NoTxn(ctx), pacta.InitiativeInvitationID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to retrieve initiative invitation", zap.Error(err))
+	}
+	result, err := conv.InitiativeInvitationToOAPI(ii)
+	if err != nil {
+		return nil, err
+	}
+	return api.GetInitiativeInvitation200JSONResponse(*result), nil
+}
+
+// Claims this initiative invitation, if it exists
+// (POST /initiative-invitation/{id})
```

Comment:
nit: I think of "claiming an invitation" as a non-CRUD action that should get a custom verb, like `POST /initiative-invitation/{id}:claim`. Benefits of this:

- Unambigious about what it's for, makes reading logs and stuff easier
- If you have another "updating an initiative invitation" endpoint you need to add, you won't have a collision on `POST /initiative-invitation/{id}`

[Some Google Cloud guide that discusses this at length](https://cloud.google.com/apis/design/custom_methods)

Reply:
Done